tests/capture/test_operators.py

```diff
@@ -0,0 +1,301 @@
+    eqn = jaxpr.eqns[0]
+
+    assert eqn.primitive == qml.PauliRot._primitive
+    assert eqn.params == {"pauli_word": "XY", "id": None, "n_wires": 2}
+
+    assert len(eqn.invars) == 3
+    assert jaxpr.jaxpr.invars == eqn.invars
+
+    qml.capture.disable_plxpr()
+
+
+class TestTemplates:
+    def test_variable_wire_non_parametrized_template(self):
+        """Test capturing a variable wire, non-parametrized template like GroverOperator."""
+
+        qml.capture.enable_plxpr()
+
+        jaxpr = jax.make_jaxpr(qml.GroverOperator)(wires=(0, 1, 2, 3, 4, 5))
+
+        assert len(jaxpr.eqns) == 1
+        eqn = jaxpr.eqns[0]
+
+        assert eqn.primitive == qml.GroverOperator._primitive
+        assert eqn.params == {"n_wires": 6}
+        assert eqn.invars == jaxpr.jaxpr.invars
+
+        qml.capture.disable_plxpr()
+
+    def test_nested_template(self):
+        """Test capturing a template that contains a nested opeartion defined outside the qfunc."""
+
+        qml.capture.enable_plxpr()
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.X(0), qml.Z(0)]
+        H = qml.dot(coeffs, ops)
+
+        def qfunc(Hi):
+            qml.TrotterProduct(Hi, time=2.4, order=2)
+
+        jaxpr = jax.make_jaxpr(qfunc)(H)
+
+        assert len(jaxpr.eqns) == 6
+
+        # due to flattening and unflattening H
+        assert jaxpr.eqns[0].primitive == qml.X._primitive
+        assert jaxpr.eqns[1].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[2].primitive == qml.Z._primitive
+        assert jaxpr.eqns[3].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[4].primitive == qml.ops.Sum._primitive
+
+        eqn = jaxpr.eqns[5]
+        assert eqn.primitive == qml.TrotterProduct._primitive
+        assert eqn.invars == jaxpr.eqns[4].outvars  # the sum op
+
+        assert eqn.params == {"order": 2, "time": 2.4}
+
+        with qml.queuing.AnnotatedQueue() as q:
+            jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, coeffs[0], coeffs[1])
+
+        assert len(q) == 1
+        assert q.queue[0] == qml.TrotterProduct(H, time=2.4, order=2)
+
+        qml.capture.disable_plxpr()
+
+
+class TestOpmath:
+    """Tests for capturing operator arithmetic."""
+
+    def test_adjoint(self):
+        """Test the adjoint on an op can be captured."""
+
+        qml.capture.enable_plxpr()
+
+        jaxpr = jax.make_jaxpr(qml.adjoint)(qml.X(0))
+
+        assert len(jaxpr.eqns) == 2
+        assert jaxpr.eqns[0].primitive == qml.X._primitive
+
+        eqn = jaxpr.eqns[1]
+        assert eqn.primitive == qml.ops.Adjoint._primitive
+        assert eqn.invars == jaxpr.eqns[0].outvars  # the pauli x op
+        assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+        assert eqn.params == {}
+
+        qml.capture.disable_plxpr()
+
+    def test_control(self):
+        """Test a nested control operation."""
+
+        qml.capture.enable_plxpr()
+
+        def qfunc(op):
+            qml.ctrl(op, control=3, control_values=[0, 0])
```

Comment:
Should there only be one control value, for one control wire?

Reply:
... yes.... I guess that's the downside of all this. We miss out on some of the valdidation. Fixing locally.